Patch updated from the below for Meson 0.55.0.

From 4db4fd79d9bb2b98cea1117f22b6c97942ab2ecd Mon Sep 17 00:00:00 2001
From: Eric Le Bihan <eric.le.bihan.dev@free.fr>
Date: Sat, 14 Jul 2018 11:18:45 +0200
Subject: [PATCH] Only fix RPATH if install_rpath is not empty

Signed-off-by: Eric Le Bihan <eric.le.bihan.dev@free.fr>
[Fix: remove leftover from original/unconditional code]
Signed-off-by: Peter Seiderer <ps.report@gmx.net>
---
 mesonbuild/minstall.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/mesonbuild/minstall.py b/mesonbuild/minstall.py
index 0be01fe..71615b4 100644
--- a/mesonbuild/minstall.py
+++ b/mesonbuild/minstall.py
@@ -512,8 +512,14 @@ class Installer:
             if file_copied:
                 self.did_install_something = True
                 try:
-                    depfixer.fix_rpath(outname, t.rpath_dirs_to_remove, install_rpath, final_path,
-                                       install_name_mappings, verbose=False)
+                    # Buildroot check-host-rpath script expects RPATH
+                    # But if install_rpath is empty, it will be stripped.
+                    # So, preserve it in this case.
+                    if install_rpath:
+                        depfixer.fix_rpath(outname, t.rpath_dirs_to_remove, install_rpath, final_path,
+                                           install_name_mappings, verbose=False)
+                    else:
+                        print("Skipping RPATH fixing")
                 except SystemExit as e:
                     if isinstance(e.code, int) and e.code == 0:
                         pass
